File: numero cifre/ncifre.py
def ncifre(val):
  # caso base
  if(val/10 < 1):
    return 1

  # passo induttivo
  else:
    return 1 + ncifre(val/10)
    
# __ main __
val = int(input("Inserisci un numero: "))
print("Ci sono " + str(ncifre(val)) + " cifre")

File: numero cifre/ncifre.c
#include <stdio.h>

// int perchè voglio restituire un intero
int ncifre(int val){
  
  // caso base
  if(val/10 < 1){
    return 1;
  }
  
  // passo induttivo
  else{
    return 1 + ncifre(val/10);
  }
  
}

int main(){

  int val;
  
  printf("Inserisci un numero: ");
  scanf("%d", &val);
  
  printf("Ci sono %d cifre\n", ncifre(val));
  
  return 0;
}